dnf-behave-tests/features/steps/transaction.py

```diff
@@ -58,6 +59,41 @@ def check_dnf_transaction(context, mode):
     lines = context.cmd_stdout.splitlines()
     dnf_transaction = parse_transaction_table(lines)
 
+    if mode == "exact_match":
+        # create lists of tuples (rows of row items) for expected and found
+        expected_lists = sorted([tuple(r) for r in context.table])
+        found_lists = []
+        for action, pkgs in dnf_transaction.items():
+            for pkg in pkgs:
+                found_lists.append((action, str(pkg)))
+        found_lists.sort()
```

Comment:
Unfortunately, we still have the "unchanged" state in some scenarios even when checking for the exact match, so this change requires to adjust these tests, too.

Reply:
Oh, and similarly, there are "absent" and "present" states. I don't think any of these states bring any value in the "exact_match" mode. Maybe in some cases they are useful for quicker understanding the test scenario, but it can be achieved using comments instead.

Reply:
I've actually dropped the transaction diff commit from the PR.

I'll try to drop the whole non-exact mode and all the stuff around it and send another PR. It is just an unnecessary complication while making the tests less reliable at the same time.